common/types/chain.go

```diff
@@ -4,9 +4,10 @@ import "fmt"
 
 // Sequence represents the base type, for any chain's sequence object.
 // It should be convertible to a string
-type Sequence interface {
+type Sequence[SEQ any] interface {
 	fmt.Stringer
 	Int64() int64 // needed for numeric sequence confirmation - to be removed with confirmation logic generalization: https://smartcontract-it.atlassian.net/browse/BCI-860
+	Next() SEQ    // needed to produce the next usable sequence for a transaction
```

Comment:
This should be Next() Sequence

Reply:
Related to the comment below but this was kept as SEQ so that the `Next` method can return itself instead of having to return `Sequence` explicitly